.github/workflows/vtk-pre-test.yml

```diff
@@ -0,0 +1,66 @@
+name: VTK Master Testing
+
+on:
+  push:
+    branches:
+      # To resolve issues on VTK master and test, use this branch name pattern
+      - 'maint/vtk-master*'
+  workflow_dispatch:
+  schedule:
+    - cron: "0 4 * * *"
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+env:
+  ALLOW_PLOTTING: true
+  SHELLOPTS: 'errexit:pipefail'
+
+jobs:
+  VTK-master:
+    name: Linux VTK Master Testing
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: [3.7, 3.8, 3.9, "3.10", "3.11"]
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 2
+
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python-version }}
+          cache: 'pip'
+          cache-dependency-path: requirements_test.txt
+
+      - uses: actions/cache@v3
+        with:
+          path: ~/.local/share/pyvista/examples
+          key: Examples-1-${{ hashFiles('*') }}
+          restore-keys: |
+            Examples-1-
+
+      - name: Install pyvista
+        run: pip install . --no-deps
+
+      - name: Install Testing Requirements
+        run: pip install -r requirements_test.txt
+
+      - name: Set up VTK with OSMesa
+        run: |
+          pip uninstall vtk -y
+          pip install vtk_osmesa --pre --no-cache --extra-index-url https://gitlab.kitware.com/api/v4/projects/13/packages/pypi/simple
```

Comment:
@akaszynski, @MatthewFlamm, @AlejandroFernandezLuces: this is the issue I am referring to in https://github.com/pyvista/pytest-pyvista/pull/41

Reply:
I think this is a separate issue from the issue you linked, but the impact is sometimes encountered in installing that package.  We have a dependency for `vtk` in pyvista, but apparently `vtk_osmesa` can be used without problem.  But, this means that `pyvista` still looks for `vtk`.  The problem is in the dependencies for `pyvista` itself, not in `pytest-pyvista` IMO.  I dont know if there is a solution for an 'either/or' dependency resolution in pip specification.

Reply:
Indeed the issue is wit `pyvista` having `vtk` in its hard requirements. This is something I think we will have a lot more friction changing. And so a simpler fix is to just remove `pyvista` from `pytest-pyvista`'s hard requirements to work around this.